plugins/com.google.cloud.tools.eclipse.appengine.standard.java8/src/com/google/cloud/tools/eclipse/appengine/standard/java8/AppEngineWebBuilder.java

```diff
@@ -86,7 +88,9 @@ private void checkRuntimeElement(IFacetedProject project, IFile appEngineWebDesc
         Set<Action> updates = new HashSet<>();
         if (hasJava8Runtime) {
           updates.add(new Action(Action.Type.VERSION_CHANGE, JavaFacet.VERSION_1_8, null));
-        } else {
+        } else if (!MavenUtils.hasMavenNature(project.getProject())) {
+          // see https://github.com/GoogleCloudPlatform/google-cloud-eclipse/issues/1941
+          // still not totally clear why this works for standard projects and not maven
```

Comment:
Chatted with @elharo. It's downright weird when a user's action to upgrade the facet version is reverted immediately against their will, so this is at least reasonable improvement.